rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -782,7 +782,11 @@ trait PromiseGeneration_buildNodeConfigurations extends PromiseGenerationService
       val nodeConfigs = bestEffort(nodeContexts.toSeq) { case (nodeId, context) =>
 
           (for {
-            drafts          <- Box(policyDraftByNode.get(nodeId)) ?~! "Promise generation algorithme error: cannot find back the configuration information for a node"
+            drafts          <- (policyDraftByNode.get(nodeId) match {
+                                  case Some(d) => Full(d)
+                                  case None    =>
+                                    Failure("Promise generation algorithme error: cannot find back the configuration information for a node")
```

Comment:
could you change the text to 
Promise generation algorithm error: cannot find back the configuration information for node ${nodeId.value}